go-controller/pkg/csrapprover/approver.go

```diff
@@ -78,51 +92,48 @@ var Predicate = predicate.Funcs{
 
 // NewController creates a new OVNKubeCSRController
 func NewController(client crclient.Client,
-	commonNamePrefixes string,
-	organization []string,
-	groups, userPrefixes sets.Set[string],
+	csrAcceptanceConditions []CSRAcceptanceCondition,
 	usages sets.Set[certificatesv1.KeyUsage],
 	maxDuration time.Duration,
 	recorder record.EventRecorder) *OVNKubeCSRController {
-	c := &OVNKubeCSRController{
-		name:               ControllerName,
-		client:             client,
-		commonNamePrefixes: commonNamePrefixes,
-		organization:       organization,
-		groups:             groups,
-		usages:             usages,
-		userPrefixes:       userPrefixes,
-		maxDuration:        maxDuration,
-		recorder:           recorder,
-	}
-	return c
-}
 
-func (c *OVNKubeCSRController) filterCSR(csr *certificatesv1.CertificateSigningRequest) bool {
-	nsn := types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}
-	csrPEM, _ := pem.Decode(csr.Spec.Request)
-	if csrPEM == nil {
-		klog.Errorf("Failed to PEM-parse the CSR block in .spec.request: no CSRs were found in %s", nsn)
-		return false
+	commonNamePrefixes := []string{}
+	// initialize Sets from slices
+	for i, v := range csrAcceptanceConditions {
+		csrAcceptanceConditions[i].groupsSet = sets.New[string](v.Groups...)
+		csrAcceptanceConditions[i].userPrefixesSet = sets.New[string](v.UserPrefixes...)
+		commonNamePrefixes = append(commonNamePrefixes, v.CommonNamePrefix)
 	}
 
-	x509CSR, err := x509.ParseCertificateRequest(csrPEM.Bytes)
-	if err != nil {
-		klog.Errorf("Failed to parse the CSR .spec.request of %q: %v", nsn, err)
-		return false
+	klog.V(5).Infof("YYY: %v", csrAcceptanceConditions)
+	return &OVNKubeCSRController{
+		name:                    ControllerName,
+		client:                  client,
+		csrAcceptanceConditions: csrAcceptanceConditions,
+		commonNamePrefixes:      commonNamePrefixes,
+		usages:                  usages,
+		maxDuration:             maxDuration,
+		recorder:                recorder,
 	}
+}
 
-	return strings.HasPrefix(x509CSR.Subject.CommonName, c.commonNamePrefixes) &&
-		csr.Spec.SignerName == certificatesv1.KubeAPIServerClientSignerName
+func (c *OVNKubeCSRController) filterCSR(csr *certificatesv1.CertificateSigningRequest, x509CSR *x509.CertificateRequest) bool {
+	matched := false
+	for _, v := range c.commonNamePrefixes {
+		if strings.HasPrefix(x509CSR.Subject.CommonName, v) {
+			matched = true
```

Comment:
You could `return csr.Spec.SignerName == certificatesv1.KubeAPIServerClientSignerName` here instead of using the `matched` var.